### DISCUSSION TITLE
DsfrDataTable - Ajout de slots pour le loading et les messages d'erreur avec des props qui vont avec

### DISCUSSION THREAD

**Discussion by GabyFumo** [dnum-mi/vue-dsfr#953] [Ideas]
Ajouter des slots permettra à l'utilisateur de savoir si ses données ont bien été récupérées ou de savoir si un problème est présent.

Il pourrait y avoir un slot par défaut pour le loading comme :
```vue
<slot
  v-if="loading"
  name="loading"
>
  <VIcon
    name="ri-loader-4-line"
    animation="spin"
    style="font-size: 3rem;"
   />
</slot>
```
Tout en gardant la possibilité de personnaliser ces parties.

Ainsi on ne laissera pas l'utilisateur dans le flou sur ce qui se passe avec la récupération des données.

**Comment by laruiss** [comment]
Oui, seulement cela soulève de nombreuses questions, comme par exemple :

- Où doit s’afficher ce loader ?
- Où doit être ce slot ? après la balise `table` ? après la balise `tbody` ?
- Est-ce qu’on fait un overlay ? sur le tbody seulement ? sur le table seulement ? sur tout le composant (caption + action en haut + actions en bas) ?
- ...
